server/src/main/java/org/opensearch/common/settings/FeatureFlagSettings.java

```diff
@@ -32,7 +32,6 @@ protected FeatureFlagSettings(
     public static final Set<Setting<?>> BUILT_IN_FEATURE_FLAGS = Set.of(
         FeatureFlags.EXTENSIONS_SETTING,
         FeatureFlags.IDENTITY_SETTING,
-        FeatureFlags.CONCURRENT_SEGMENT_SEARCH_SETTING,
```

Comment:
If this setting is removed, then the cluster will fail to start up with an "IllegalArgumentException: unknown setting" error if the feature flag is still present in the yml configuration.

@reta I believe in the past we haven't handled this case, meaning users would have to remove the feature flag upon upgrade. What do you think we should do here? I think allowing "enabled = true" with a deprecation warning is probably a better experience, but also there are no explicit guarantees with experimental features. I'm less sure what to do if "enable = false" is present in the config because the feature flag will no longer disable the feature.

Reply:
> What do you think we should do here? 

I think we should be following a path similar to settings archiving:
 - do not fail, and
 - print deprecation warning

It looks pretty unfriendly to fail the service just because the experimental feature flag is gone.

Reply:
Looking at https://github.com/opensearch-project/OpenSearch/blob/main/server/src/main/java/org/opensearch/action/admin/cluster/settings/SettingsUpdater.java#L73-L92 it looks like there's already some existing logic for handling this case for cluster settings so we would need to extend that to support feature flags as well.

I think that can be handled as a separate improvement outside of the scope of this PR though. Unfortunately we already have [many feature flags](https://github.com/search?q=repo%3Aopensearch-project%2FOpenSearch+Remove+Feature+Flag&type=commits) that have been removed in the past and all of them are susceptible to this problem. As @andrross mentioned there are no explicit guarantees with experimental features today (for example what if we need to change the name of an experimental feature?).

Reply:
@jed326 sure, I think the feature flag management is long overdue ....

Reply:
> @jed326 sure, I think the feature flag management is long overdue ....

Let me create a new issue for this shortly with our discussion here